pkg/deprecation/deprecation.go

```diff
@@ -0,0 +1,43 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package deprecation
+
+type Warning string
+
+const (
+	// Prefix is a standard prefix for all Warnings, used for filtering plugin Exports
+	Prefix = "io.containerd.deprecation/"
+	// PullSchema1Image is a warning for the use of schema 1 images
+	PullSchema1Image Warning = Prefix + "pull-schema-1-image"
+)
+
+var messages = map[Warning]string{
+	PullSchema1Image: "Schema 1 images are deprecated since containerd v1.7 and removed in containerd v2.0. " +
+		`Since containerd v1.7.8, schema 1 images are identified by the "io.containerd.image/converted-docker-schema1" label.`,
+}
```

Comment:
Why don't we need GoPluginLibrary in 1.7?

Reply:
GoPluginLibrary is first deprecated in 2.0 with a target removal in 2.1.  It's not deprecated in 1.7.

Reply:
We _could_ warn about it in previous versions, but it's less clear whether that's the right choice since it somewhat dilutes the message of the warnings overall.  (For the Go plugin library case it also means some slightly-more invasive changes to the place plugins are loaded.  I did this in `main` and it's not _that_ risky, but I was avoiding it here since they're not actually deprecated in 1.7.

Reply:
Oh I misunderstood the purpose of the warnings then. I thought we wanted to pre-warn the 1.6 users about the upcoming 2.0 deprecations. 

Have we considered that, since 1.7 is not an LTS so users might update from 1.6 to the next LTS, which will be after 2.0?

Reply:
> I thought we wanted to pre-warn the 1.6 users about the upcoming 2.0 deprecations.

We want to pre-warn users that they are using deprecated functionality that will be removed.  Go library plugins are not removed in 2.0 (they're deprecated in 2.0, removed in 2.1).

> Have we considered that, since 1.7 is not an LTS so users might update from 1.6 to the next LTS, which will be after 2.0?

Yeah, this is where it gets a bit tricky.  Schema 1 images are first deprecated in _1.7_, so if we don't produce a warning in _1.6_ then LTS users who upgrade may be left out.

I think you've convinced me that we should backport all known deprecations rather than just the features that are deprecated in a specific version.  I'll update this PR to handle the Go library plugins as well.

Reply:
@ruiwen-zhao Updated, PTAL.